### PR TITLE
Added workload pod annotations to rendered manifests.

### LINKED ISSUE #562: Introduce pod annotation configuration
```yaml
services:
    x-k8s:
        disabled: false
        workload:
            ...
            annotations: # pod annotations - this is a new concept which we currently don't support
              key1: val1
              key2: val2
```

### PR DESCRIPTION
**Pull request by ezodude**
Resolves #562 

- Adds annotations to rendered manifests as per configured workload annotations.
- Updates the documentation to include annotations config.

Example rendered deployment:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    service: wordpress
  name: wordpress
spec:
  replicas: 25
  selector:
    matchLabels:
      service: wordpress
  strategy:
  ...
  template:
    metadata:
      annotations:
        hello: world
      creationTimestamp: null
      labels:
        network/default: "true"
        service: wordpress
    spec:
       ...
status: {}
```

### CODE REVIEW

**Review by marcinc — COMMENTED** [review]
LGTM. Once reference is updated we're good to go with it.

File: pkg/kev/converter/kubernetes/transform.go [hunk 1]
```diff
@@ -1668,7 +1668,7 @@ func (k *Kubernetes) updateKubernetesObjects(projectService ProjectService, obje
 	// @step add PVCs to objects
 	// Looping on the slice pvcs instead of `*objects = append(*objects, pvcs...)`
 	// because the type of objects and pvcs is different, but when doing append
-	// one element at a time it gets converted to runtime.Object for objects slice
+	// one element at a time it getfs converted to runtime.Object for objects slice
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/transform.go:1671]: small typo

File: pkg/kev/config/svcext.go [hunk 2]
```diff
@@ -415,17 +415,18 @@ func validateDNSSubdomainNameIfAny(fl validator.FieldLevel) bool {
...
-	RestartPolicy         RestartPolicy  `yaml:"restartPolicy,omitempty" validate:"restartPolicy"`
-	ImagePull             ImagePull      `yaml:"imagePull,omitempty"`
-	Resource              Resource       `yaml:"resource,omitempty"`
-	Autoscale             Autoscale      `yaml:"autoscale,omitempty"`
-	PodSecurity           PodSecurity    `yaml:"podSecurity,omitempty"`
+	Type                  WorkloadType      `yaml:"type,omitempty" validate:"workloadType"`
+	Replicas              int               `yaml:"replicas" validate:""`
+	ServiceAccountName    string            `yaml:"serviceAccountName,omitempty" validate:"subdomainIfAny"`
+	RollingUpdateMaxSurge int               `yaml:"rollingUpdateMaxSurge,omitempty" validate:""`
+	PodAnnotations        map[string]string `yaml:"annotations,omitempty"`
```
Comment (marcinc) [inline pkg/kev/config/svcext.go:422]: Let's stick to `Annotations` here. We know that annotations will be on the pod spec internally.

File: pkg/kev/converter/kubernetes/transform_test.go [hunk 3]
```diff
@@ -863,7 +935,7 @@ var _ = Describe("Transform", func() {
 			})
 		})
 
-		When("project service extension instructing to expose the k8s service with domain and annotations", func() {
+		When("project service extension instructing to expose the k8s service with domain and podAnnotations", func() {
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/transform_test.go:938]: I suppose here we're dealing with IngressAnnotations so the context description is wrong here and below.

**Review by marcinc — COMMENTED** [review]
left a few minor comments.

File: pkg/kev/converter/kubernetes/transform_test.go (hunk 3 above)
Comment (ezodude) [reply pkg/kev/converter/kubernetes/transform_test.go:938]: That was a find and replace gone bad. Will fix.

File: pkg/kev/converter/kubernetes/transform.go [hunk 4]
```diff
@@ -1339,10 +1339,10 @@ func (k *Kubernetes) configEnvs(projectService ProjectService) ([]v1.EnvVar, err
 			})
 		case "pod":
 			// Selects a field of the pod
-			// supported paths: metadata.name, metadata.namespace, metadata.labels, metadata.annotations,
+			// supported paths: metadata.name, metadata.namespace, metadata.labels, metadata.podAnnotations,
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/transform.go:1342]: I think this is a leftover?  `metadata.podAnnotations` -> `metadata.annotations` here and in paths below.

File: pkg/kev/converter/kubernetes/project_service.go [hunk 5]
```diff
@@ -130,7 +139,7 @@ func (p *ProjectService) tlsSecretName() string {
 	return p.SvcK8sConfig.Service.Expose.TlsSecret
 }
 
-// ingressAnnotations returns the ingress annotations for exposed service (to be used in the ingress configuration)
+// ingressAnnotations returns the ingress podAnnotations for exposed service (to be used in the ingress configuration)
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/project_service.go:142]: podAnnotations -> annotations

File: pkg/kev/converter/kubernetes/transform.go [hunk 6]
```diff
@@ -1684,8 +1684,8 @@ func (k *Kubernetes) updateKubernetesObjects(projectService ProjectService, obje
 	// @step configure capabilities
 	capabilities := k.configCapabilities(projectService)
 
-	// @step configure annotations
-	annotations := configAnnotations(projectService)
+	// @step configure podAnnotations
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/transform.go:1687]: small change `podAnnotations` -> `pod annotations`

File: pkg/kev/converter/kubernetes/transform_test.go [hunk 7]
```diff
@@ -532,6 +532,30 @@ var _ = Describe("Transform", func() {
 				Expect(d.Spec.Strategy.RollingUpdate.MaxUnavailable.IntValue()).To(Equal(0))
 			})
 		})
+
+		Context("for project service configured with pod annotations", func() {
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/transform_test.go:536]: I guess we can just say `with annotations`? the cases below will explain what's the expected behaviour. The same comment applies to the other specs where we use `pod annotations` in context description.

File: pkg/kev/converter/kubernetes/transform_test.go [hunk 8]
```diff
@@ -874,7 +946,7 @@ var _ = Describe("Transform", func() {
 				projectService.SvcK8sConfig.Service.Expose.Domain = "domain.name"
 			})
 
-			It("initialises Ingress with configured ingress annotations", func() {
+			It("initialises Ingress with configured ingress podAnnotations", func() {
```
Comment (marcinc) [inline pkg/kev/converter/kubernetes/transform_test.go:949]: `podAnnotations` -> `annotations`

**Review by marcinc — COMMENTED** [review]
a few more minor comments :)

File: pkg/kev/converter/kubernetes/transform.go (hunk 4 above)
Comment (ezodude) [reply pkg/kev/converter/kubernetes/transform.go:1342]: Yup.

File: pkg/kev/converter/kubernetes/transform.go (hunk 6 above)
Comment (ezodude) [reply pkg/kev/converter/kubernetes/transform.go:1687]: Actually this should just say `annotations` as we don't configure pod annotations  here.

**Review by marcinc — APPROVED** [review]
LGTM 👍

**Pull request merged**